src/python/pants/init/engine_initializer.py

```diff
@@ -366,6 +369,13 @@ def symbol_table_singleton() -> SymbolTable:
     def union_membership_singleton() -> UnionMembership:
       return UnionMembership(build_configuration.union_rules())
 
+    @rule
+    def single_build_file_address(specs: Specs) -> BuildFileAddress:
+      build_file_addresses = yield Get(BuildFileAddresses, Specs, specs)
+      if len(build_file_addresses.dependencies) != 1:
+        raise Exception("This goal only works with a single target, but has been given multiple targets.")
```

Comment:
This should probably mention which `Specs` objects we parsed/matched... could probably use https://github.com/pantsbuild/pants/blob/a0bcc2314c486fcaa34de9401c29890799ead1ba/src/python/pants/engine/build_files.py#L204-L214